features/user-profile.md

```diff
@@ -0,0 +1,226 @@
+# API Design Overview
+
+This document introduces the new Skygear User Record (Profile), which is
+re-designed from user record and user object, utilizing user profile discovery.
+The design depends on [the field-based ACL](field-based-acl.md) feature.
+
+## New User Record
+
+The newly designed user record contains different information of user.By
+setting field-based ACL, user record can serve as both public profile and
+private profile.
+
+The following table gives an example of the field-based ACL of user record
+fulfilling: 1) both username, email and status are readable and discoverable;
+2) profile photo is readable, but not queryable; 3) game score is queryable;
+and 4) birthday is private:
+
+| Class | UserRole |    Field    | AccessLevel | DiscoveryLevel |
+|-------|----------|-------------|-------------|----------------|
+| *     | Public   | *           | ReadWrite   | Queryable      |
+| User  | Owner    | *           | ReadWrite   | Queryable      |
+| User  | AnyUser  | username    | ReadOnly    | Discoverable   |
+| User  | AnyUser  | email       | ReadOnly    | Discoverable   |
+| User  | AnyUser  | status      | ReadOnly    | Discoverable   |
+| User  | AnyUser  | profile_pic | ReadOnly    | NotQueryable   |
+| User  | AnyUser  | game_score  | ReadOnly    | Queryable      |
+| User  | AnyUser  | birthday    | NoAccess    | NotQueryable   |
+
+## Authentication Object
+
+The new authentication object (i.e. `_auth` object) is to replace the old
+ `_user` object, to avoid ambiguity with the user record.
```

Comment:
Also create a view called `_user` iff migrating the schema from an old version. The `_user` view should resemble the `_user` table in previous schema version.